src/main/java/io/kestra/plugin/gcp/runner/GcpCloudRunTaskRunner.java

```diff
@@ -0,0 +1,409 @@
+        final GoogleCredentials credentials = CredentialService.credentials(runContext, this);
+
+        boolean hasFilesToUpload = !ListUtils.isEmpty(filesToUpload);
+        if (hasFilesToUpload && bucket == null) {
+            throw new IllegalArgumentException("You must provide a Cloud Storage Bucket to use `inputFiles` or `namespaceFiles`");
+        }
+        boolean hasFilesToDownload = !ListUtils.isEmpty(filesToDownload);
+        boolean outputDirectoryEnabled = taskCommands.outputDirectoryEnabled();
+
+        if ((hasFilesToDownload || outputDirectoryEnabled) && bucket == null) {
+            throw new IllegalArgumentException("You must provide a Cloud Storage Bucket to use `outputFiles` or `{{ outputDir }}`");
+        }
+
+        Map<String, Object> additionalVars = this.additionalVars(runContext, taskCommands);
+
+        final Path workingDir = (Path) additionalVars.get(ScriptService.VAR_WORKING_DIR);
+        final Path outputDir = (Path) additionalVars.get(ScriptService.VAR_OUTPUT_DIR);
+
+        String workingDirectoryToBlobPath = MOUNT_PATH.relativize(workingDir).toString();
+        boolean hasBucket = this.bucket != null;
+
+        if (hasFilesToUpload || outputDirectoryEnabled) {
+            GcsUtils.of(renderedProjectId, credentials).uploadFiles(runContext,
+                filesToUpload,
+                renderedBucket,
+                toAbsoluteBlobPathWithoutMount(workingDir),
+                toAbsoluteBlobPathWithoutMount(outputDir),
+                outputDirectoryEnabled
+            );
+        } else {
+            // Otherwise, create a blob for the WORKING_DIR to be able to mount it in the container volume.
+            BlobInfo workingDirBlob = BlobInfo.newBuilder(BlobId.of(renderedBucket, workingDirectoryToBlobPath + "/")).build();
+            try (Storage storage = GcsUtils.of(renderedProjectId, credentials).storage(runContext)) {
+                storage.create(workingDirBlob);
+            }
+        }
+
+        try (JobsClient jobsClient = JobsClient.create(JobsSettings.newBuilder().setCredentialsProvider(() -> credentials).build());
+             ExecutionsClient executionsClient = ExecutionsClient.create(ExecutionsSettings.newBuilder().setCredentialsProvider(() -> credentials).build());
+             Logging logging = LoggingOptions.getDefaultInstance().toBuilder().setCredentials(credentials).build().getService()) {
+
+            // Create new Job TaskTemplate
+            TaskTemplate.Builder taskBuilder = TaskTemplate.newBuilder();
+
+            // Create and add Volume
+            Volume volume = Volume.newBuilder()
+                .setGcs(
+                    GCSVolumeSource
+                        .newBuilder()
+                        .setReadOnly(false)
+                        .setBucket(renderedBucket)
+                        .build()
+                )
+                .setName(VOLUME_NAME)
+                .build();
+            taskBuilder.addVolumes(volume);
```

Comment:
You should only create a volume if needed, otherwise the user must always provide a bucket even if not needed